Automattic/jetpack
# [RNMobile] Add React Native friendly entry points for a few packages

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5357

#### Changes proposed in this Pull Request:

PR #27937 added a few changes that broke the mobile editor build.

There are two issues:
### entry point resolution 
The Metro JS bundler can not resolve index files via the `exports` field in package.json. The entry point needs to be in the package root or specified in the `main` field.

### SCSS compliation.
Metro JS was unable to compile the scss files in the `js-packages/connection` package. The issue is not universal for Metro but errors are hard to resolve in the `connection` components.

This PR adds:
- A `main` field to the `js-packages/config` package to point to `./src/index.js` . Per the[ Node docs](https://nodejs.org/api/packages.html#main), this should not impact the `exports` field. 
#### Other information:

- A native only index.js file to `js-pacakges/connection` to only export native required modules.


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






- Verify that the mobile build is working (see https://github.com/wordpress-mobile/gutenberg-mobile/pull/5366 )
